### PR TITLE
[Performance] Support NCCL backed for sparse embedding

### PR DESCRIPTION
**Pull request by isratnisa**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/model/embed.py [hunk 1]
```diff
@@ -208,17 +209,15 @@ def __init__(self,
...
-        # TODO (Israt): Add NCCL support to distDGL.
-        if is_distributed() and get_backend() == "nccl":
-            if self.use_node_embeddings:
-                raise NotImplementedError('NCCL backend is not supported for utilizing \
-                    node embeddings. Please use gloo backend.')
-            for ntype in g.ntypes:
-                if not feat_size[ntype]:
-                    raise NotImplementedError('NCCL backend is not supported for utilizing \
-                        learnable embeddings on featureless nodes. Please use gloo backend.')
+        if dgl.__version__ <= "1.1.2":
```
Comment (zheng-da) [inline python/graphstorm/model/embed.py:212]: can you merge the if condition with the next if condition?

Comment (isratnisa) [reply python/graphstorm/model/embed.py]: Done

File: python/graphstorm/model/embed.py [hunk 2]
```diff
@@ -208,17 +209,14 @@ def __init__(self,
...
 
-        # NCCL backend is not supported for utilizing learnable embeddings on nodes. It has a
-        # dependency on distDGL (PR https://github.com/dmlc/dgl/pull/5929).
-        # TODO (Israt): Add NCCL support to distDGL.
-        if is_distributed() and get_backend() == "nccl":
+        if dgl.__version__ <= "1.1.2" and is_distributed() and get_backend() == "nccl":
             if self.use_node_embeddings:
-                raise NotImplementedError('NCCL backend is not supported for utilizing \
-                    node embeddings. Please use gloo backend.')
+                raise NotImplementedError('NCCL backend is not supported for utilizing ' +
```
Comment (zheng-da) [inline python/graphstorm/model/embed.py:214]: can you also mention that people can potentially use a newer version of DGL?

Comment (isratnisa) [reply python/graphstorm/model/embed.py:214]: Added in the Error comment. Does additional comments required?

**Review by zheng-da — APPROVED** [review]

**Pull request merged**